src/scripts/main.js

```diff
@@ -1 +1,51 @@
 'use strict';
+
+const clicks = ['click', 'contextmenu'];
+
+const firstPromise = new Promise((resolve, reject) => {
+  setTimeout(() => reject(new Error(`First promise was rejected`)), 3000);
+
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  clicks.forEach((e) => {
+    document.addEventListener(e, () => {
+      resolve('Second promise was resolved');
+    });
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  clicks.forEach((e) => {
+    document.addEventListener(e, () => {
+      clicks.splice(clicks.indexOf(e), 1);
+
+      if (clicks.length === 0) {
+        resolve('Third promise was resolved');
+      }
+    });
+  });
```

Comment:
The `thirdPromise` is designed to resolve after both left and right clicks have occurred, which matches the task requirements.